rtrlib/rtr_mgr.h

```diff
@@ -67,28 +68,37 @@ struct rtr_mgr_group {
 	enum rtr_mgr_status status;
 };
 
+//TODO Find a nicer way todo a linked list (without writing our own)
+struct rtr_mgr_group_node {
+	tommy_node node;
+	struct rtr_mgr_group *group;
+};
+
 typedef void (*rtr_mgr_status_fp)(const struct rtr_mgr_group *,
 				  enum rtr_mgr_status,
 				  const struct rtr_socket *,
 				  void *);
 
+//TODO Add refresh, expire, and retry intervals to config for easier access.
 struct rtr_mgr_config {
-	struct rtr_mgr_group *groups;
+	tommy_list groups;
```

Comment:
It would be great to hide the actual list implementation from the user. This would allow to change freely the data structure in the background without breaking the API.
I also don't like the idea much that the user has additional to know the tommy_list API to use the rtrlib.

The `struct rtr_mgr_group` should be a opaque data structure that is only accessible via helper functions, something like:
```
struct rtr_mgr_group
struct rtr_mgr_group *rtr_group_alloc();
void rtr_group_free(struct rtr_mgr_group *);
int rtr_mgr_group_append(struct rtr_mgr_group *list, struct rtr_mgr_group *new);
void rtr_mgr_group_remove(struct rtr_mgr_group *grp);


void rtr_mgr_group_add_socket(struct rtr_mgr group *, struct rtr_socket *);
void rtr_mgr_group_set_pref(struct rtr_mgr group *, uint8_t pref);
...
```
Refactoring that API might be a separate topic. The changes in the this pull request are consistent with the previous API but it makes it even more complex :-)